### PR TITLE
Add vsphere-cpi e2e test as a presubmit job.

### PR DESCRIPTION
**Pull request by HanFa**
Create a presubmit job for cloud-provider-vsphere e2e.

### CODE REVIEW

File: hack/tools.go [hunk 1]
```diff
@@ -1,3 +1,4 @@
+//go:build tools
```
Comment (lubronzhan) [inline hack/tools.go:1]: Do we need this because you are using old version golagn?

Comment (HanFa) [reply hack/tools.go]: those changes are not expected. I will revert them back

File: config/jobs/kubernetes/cloud-provider-vsphere/cloud-provider-vsphere-config.yaml [hunk 2]
```diff
@@ -246,6 +246,39 @@ presubmits:
       testgrid-tab-name: pr-verify-test-coverage
       description: Shows the test coverage of the Golang sources
 
+  # Executes the e2e tests.
+  - name: pull-cloud-provider-vsphere-e2e-test
+    decorate: true
+    labels:
+      preset-dind-enabled: "true"
+      preset-cluster-api-provider-vsphere-e2e-config: "true"
```
Comment (lubronzhan) [inline config/jobs/kubernetes/cloud-provider-vsphere/cloud-provider-vsphere-config.yaml:254]: Could you add a comment here, noting that we are borrowing credentials from CAPV job

Comment (HanFa) [reply config/jobs/kubernetes/cloud-provider-vsphere/cloud-provider-vsphere-config.yaml:254]: sure

**Pull request merged**